tx/textual/valuerenderer/coins.go

```diff
@@ -3,14 +3,19 @@ package valuerenderer
 import (
 	"context"
 	"fmt"
+	"strings"
 
 	"google.golang.org/protobuf/reflect/protoreflect"
 
 	bankv1beta1 "cosmossdk.io/api/cosmos/bank/v1beta1"
 	basev1beta1 "cosmossdk.io/api/cosmos/base/v1beta1"
 	corecoins "cosmossdk.io/core/coins"
+	"cosmossdk.io/math"
+	"cosmossdk.io/tx/textual/internal/utils"
 )
 
+const emptyCoins = "zero"
+
 // NewCoinsValueRenderer returns a ValueRenderer for SDK Coin and Coins.
 func NewCoinsValueRenderer(q CoinMetadataQueryFn) ValueRenderer {
```

Comment:
Let's add a `fd protoreflect.FieldDescriptor` field to this function, as well as a private field `fd`. (see below why)